src/repl/scala/tools/nsc/interpreter/LoopCommands.scala

```diff
@@ -24,40 +24,110 @@ class ProcessResult(val line: String) {
   override def toString = "`%s` (%d lines, exit %d)".format(line, buffer.size, exitCode)
 }
 
-trait LoopCommands {
+trait LoopCommands { self: { def echo(msg: String): Unit } =>
```

Comment:
Is there a nominal type we could use instead? The current type means `self.echo` is a reflective call.